lib/native/source/libboolector/interface_wrap.c

```diff
@@ -3519,56 +3556,57 @@ SWIGEXPORT jstring JNICALL Java_org_sosy_1lab_java_1smt_solvers_boolector_BtorJN
 	jstring jresult = 0;
 	Btor *arg1 = (Btor *) 0 ;
     char *result = 0 ;
-	char *filename = "tempdump.txt";
+	char filename[] = "boolector_help_dump_node_smt2_tempinfile-XXXXXX";
+	char *tfnwd = addTemppathToFilename(filename);
+    int fd = -1;
+
+    fd = mkstemp(tfnwd);
+    if(fd == -1) {
+      perror("ERROR CREATING TEMPORARY FILE FOR");
+    }
 	
 	(void)jenv;
     (void)jcls;
 	arg1 = *(Btor **)&jarg1; 
 	
 	char * buffer = 0;
     long length;
-  BoolectorNode *arg2 = (BoolectorNode *) 0 ;
-  arg2 = *(BoolectorNode **)&jarg2; 
+    BoolectorNode *arg2 = (BoolectorNode *) 0 ;
+    arg2 = *(BoolectorNode **)&jarg2; 
 	
     FILE *f = 0;
-    f = fopen(filename, "w+");
+    f = fdopen(fd, "w+");
 	if(f==NULL) {
 	  perror("ERROR: COULDNT DUMP NODE BECAUSE IT COULDNT CREATE A DUMP FILE");	
 	}
 
-	//fclose (f);
     
     //write
     boolector_dump_smt2_node(arg1, f, arg2);
     
     //read
-    if (f)
-    {
+    if (f) {
      fseek (f, 0, SEEK_END);
       length = ftell (f);
       fseek (f, 0, SEEK_SET);
-     buffer = malloc (length);
+     buffer = malloc (length);		//We leak this memory at the moment!
```

Comment:
Why do we need to leak memory? Can we free it at the end of this method?

Reply:
Fixed in next commit.